web/src/filter.ts

```diff
@@ -402,7 +402,8 @@ export class Filter {
             return orig_terms;
         }
 
-        const updated_terms = [...orig_terms];
+        const updated_terms = orig_terms.filter((term: NarrowTerm) => term.operator !== "dm");
```

Comment:
This function is kinda a mess; Is it even possible to have a `dm` operator here? 

I think this code path should be rewritten to just pull off any channel/topic terms and add the correct new ones, rather than having this temporary `""` operands approach, which just feels very fragile.

But that can probably be a follow-up PR; this new code does not seem obviously worse.

Reply:
>This function is kinda a mess; Is it even possible to have a dm operator here?

This comes up when we have a dm narrow with a with term pointing to a stream message (the dm term would be removed and channel-topic terms would be added here).